packages/opentelemetry-cloud-trace-exporter/src/trace.ts

```diff
@@ -95,49 +85,64 @@ export class TraceExporter implements SpanExporter {
    * service.
    * @param spans
    */
-  private _batchWriteSpans(spans: SpansWithCredentials) {
+  private _batchWriteSpans(spans: NamedSpans): Promise<ExportResult> {
     this._logger.debug('Google Cloud Trace batch writing traces');
+    // Always resolve with the ExportResult code
+    return new Promise(async resolve => {
+      if (!this._traceServiceClient) {
+        try {
+          this._traceServiceClient = await this._getClient();
+        } catch (err) {
+          err.message = `authorize error: ${err.message}`;
+          this._logger.error(err.message);
+          return resolve(ExportResult.FAILED_NOT_RETRYABLE);
+        }
+      }
 
-    return new Promise((resolve, reject) => {
-      // @todo Consider to use gRPC call (BatchWriteSpansRequest) for sending
-      // data to backend :
-      // https://cloud.google.com/trace/docs/reference/v2/rpc/google.devtools.
-      // cloudtrace.v2#google.devtools.cloudtrace.v2.TraceService
-      TraceExporter._cloudTrace.projects.traces.batchWrite(
+      const metadata = new grpc.Metadata();
+      metadata.add('x-opentelemetry-outgoing-request', '1');
+      this._traceServiceClient.BatchWriteSpans(
```

Comment:
Looks to me like this is the only part of that needs to be converted into a promise, and the above code can be moved outside of the `new Promise(...)`? Maybe you can use [`util.promisifiy()`](https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original)

 Too bad grpc doesn't support promises directly 🙁